src/ObjectStore/v1/Models/MetadataTrait.php

```diff
@@ -13,9 +13,12 @@ public function parseMetadata(ResponseInterface $response): array
         $metadata = [];
 
         foreach ($response->getHeaders() as $header => $value) {
-            if (0 === strpos($header, static::METADATA_PREFIX)) {
+            if (0 === stripos($header, static::METADATA_PREFIX)) {
                 $name            = substr($header, strlen(static::METADATA_PREFIX));
                 $metadata[$name] = $response->getHeader($header)[0];
+                if ($name !== strtolower($name)) {
```

Comment:
This if is not needed. And it makes the code slower as `strtolower` executed twice.